ext/pgsql/pgsql.c

```diff
@@ -3289,18 +3342,18 @@ PHP_FUNCTION(pg_escape_string)
 			link = FETCH_DEFAULT_LINK();
 			break;
 		default:
-			if (zend_parse_parameters(ZEND_NUM_ARGS(), "rS", &pgsql_link, &from) == FAILURE) {
+			if (zend_parse_parameters(ZEND_NUM_ARGS(), "OS", &pgsql_link, pgsql_link_ce, &from) == FAILURE) {
 				RETURN_THROWS();
 			}
-			link = Z_RES_P(pgsql_link);
+			link = Z_PGSQL_LINK_P(pgsql_link);
+            CHECK_PGSQL_LINK(link);
 			break;
 	}
 
 	to = zend_string_safe_alloc(ZSTR_LEN(from), 2, 0, 0);
+	//  TODO When can it ben null?
```

Comment:
If you the default handle is a persistent connection?

Reply:
This comment is outdated as I found out the case afterwards :) The `link` can be null when a default connection haven't been added yet.